app/util/Logger.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package util
+
+import cats.implicits.catsSyntaxEq
+import play.api.{Logger => PlayLogger}
+import play.api.libs.json.{JsValue, Json}
+import play.api.mvc.RequestHeader
+import ssttparrangement.SubmissionError
+import uk.gov.hmrc.http.CookieNames
+import uk.gov.hmrc.selfservicetimetopay.models.{PaymentPlanRequest, TTPArrangement}
+import util.RequestSupport.hc
+
+class Logger(reference: String, inClass: Class[_]) extends BaseLogger(inClass) {
+  val log: play.api.Logger = PlayLogger(reference)
+}
+
+abstract class BaseLogger(inClass: Class[_]) {
+
+  val log: play.api.Logger
+
+  def debug(message: => String)(implicit request: RequestHeader): Unit = logMessage(message, Debug)
+
+  def info(message: => String)(implicit request: RequestHeader): Unit = logMessage(message, Info)
+
+  def warn(message: => String)(implicit request: RequestHeader): Unit = logMessage(message, Warn)
+
+  def error(message: => String)(implicit request: RequestHeader): Unit = logMessage(message, Error)
+
+  def debug(message: => String, ex: Throwable)(implicit request: RequestHeader): Unit = logMessage(message, ex, Debug)
+
+  def info(message: => String, ex: Throwable)(implicit request: RequestHeader): Unit = logMessage(message, ex, Info)
+
+  def warn(message: => String, ex: Throwable)(implicit request: RequestHeader): Unit = logMessage(message, ex, Warn)
+
+  def error(message: => String, ex: Throwable)(implicit request: RequestHeader): Unit = logMessage(message, ex, Error)
+
+  protected def source = s"[class: $inClass]"
+
+  protected def context(implicit rh: RequestHeader) = s"$request $sessionId $requestId $referer $deviceId"
```

Comment:
this seems excessive - we get this info automatically anyway as other fields in kibana logs

Reply:
removed `sessionId` and `requestId` which are kibana log fields. Removed `deviceId` as excessive. Let `request` and `referer` as not in kibana log fields (that are in the request fields but useful to have them visible in the logs